src/main/java/com/nutomic/syncthingandroid/util/AsyncState.java

```diff
@@ -0,0 +1,30 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.os.AsyncTask;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Tracks the current state of a running AsyncTask without keeping a reference to it.
+ * This allows the object to be GCed as soon as it is finished. Releasing the resources
+ * allocated by the object.
+ */
+public class AsyncState {

```

Comment:
Are you sure this works, and the AsyncTask won't be GC'd while it is being executed?